clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -386,4 +386,51 @@ def ASTFunctionDeclAttr : ASTDecl<"FunctionDecl", "fndecl">;
 def ASTVarDeclAttr : ASTDecl<"VarDecl", "vardecl">;
 def ASTRecordDeclAttr : ASTDecl<"RecordDecl", "recdecl">;
 
+
+//===----------------------------------------------------------------------===//
+// ExtraFuncAttr
+//===----------------------------------------------------------------------===//
+
+def ExtraFuncAttr : AttrDef<CIR_Dialect, "ExtraFuncAttributes"> {
```

Comment:
Why is this not a `CIR_Attr`?

Reply:
CIR_Attr has a mnemonic which enforces the printing of it followed by a custom printer. So instead of  
`
cir.func @_Z2s2ii(} extra( {cir.inline = #cir.inline<always>} )`

we would have 

`
cir.func @_Z2s2ii(} cir.ExtraFuncAttributes = custom_printer`, even if the attribute is empty.

This is not as customized as we want, which prints nothing when empty.